### PR TITLE
Add root key package types and stub for Parse 

### PR DESCRIPTION
**Pull request by jw-msft**

### CODE REVIEW

File: src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_utils.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * @brief rootkeypackage_utils interface.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+#ifndef ROOTKEYPACKAGE_UTILS_H
+#define ROOTKEYPACKAGE_UTILS_H
+
+#include "aduc/rootkeypackage_types.h"
```
Comment (Nox-MSFT) [inline src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_utils.h:12]: nit: sort header

File: src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_types.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#include <azure_c_shared_utility/strings.h>
+#include <azure_c_shared_utility/vector.h>
+#include <sys/time.h> // for struct timespec
+
+/**
+ * @brief The root key keytype.
+ */
+typedef enum tagADUC_RootKey_KeyType
+{
+    RSA, /**< RSA keytype. */
```
Comment (Nox-MSFT) [inline src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_types.h:22]: Add EC?

**Review by Nox-MSFT — APPROVED** [review]

File: src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_types.h (hunk 2 above)
Comment (jw-msft) [reply src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_types.h:22]: Leaving it out for MvP. Crypto agility to add it is an orthogonal effort.

File: src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_utils.h (hunk 1 above)
Comment (jw-msft) [reply src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_utils.h:12]: It's already clang-formatted.  If I touch/change it or swap the #include lines, clang-format sorts by putting `#include < ... >` after #include " ... "

File: src/utils/rootkeypackage_utils/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+compileasc99 ()
+
+find_package (azure_c_shared_utility REQUIRED)
+
+add_library (${PROJECT_NAME} STATIC src/rootkeypackage_utils.c)
+add_library (aduc::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+
+target_include_directories (${PROJECT_NAME} PUBLIC inc ${ADUC_EXPORT_INCLUDES})
+
+target_link_libraries (${PROJECT_NAME}
```
Comment (nihemstr) [inline src/utils/rootkeypackage_utils/CMakeLists.txt:14]: Don't we need to include the library for the result.h file?

File: src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_types.h [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * @brief The root key SHA algorithms.
+ */
+typedef enum tagADUC_RootKeyShaAlgorithm
+{
+    SHA256, /**< The SHA256 algorithm. */
+    SHA384, /**< The SHA384 algorithm. */
+    SHA512, /**< The SHA512 algorithm. */
+} ADUC_RootKeyShaAlgorithm;
```
Comment (nihemstr) [inline src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_types.h:33]: We should check with Dave/PRSS to make sure we're not planning on moving to 1024/2048+

File: src/utils/rootkeypackage_utils/CMakeLists.txt (hunk 3 above)
Comment (jw-msft) [reply src/utils/rootkeypackage_utils/CMakeLists.txt:14]: result.h is in src/inc/aduc/result.h so handled by
target_include_directories (${target_name} PUBLIC inc ${ADUC_EXPORT_INCLUDES})
I may be missing something

**Pull request merged**